Tests/SKCoreTests/TaskSchedulerTests.swift

```diff
@@ -354,8 +353,9 @@ fileprivate extension TaskScheduler<ClosureTaskDescription> {
       body,
       dependencies: dependencies
     )
+    let queuedTask = await self.schedule(priority: priority, taskDescription)
     return Task(priority: priority) {
-      await self.schedule(priority: priority, taskDescription).waitToFinishPropagatingCancellation()
+      await queuedTask.waitToFinishPropagatingCancellation()
     }
   }
```

Comment:
Worth a comment?

Reply:
Added one. It was somewhat obvious in hindsight but that might also be because I’ve thought about `Task` execution order quite a bit over the last month 😄

Reply:
Yeah it wasn't immediately obvious to me 😅